src/output.rs

```diff
@@ -21,7 +21,18 @@ use crate::html;
 
 fn get_target_output_writable(output_file: Option<&str>) -> Box<dyn Write> {
     let write_target: Box<dyn Write> = match output_file {
-        Some(filename) => Box::new(File::create(filename).unwrap()),
+        Some(filename) => {
+            let output = PathBuf::from(filename);
+            if output.is_dir() {
+                panic!(
+                    "The output file {} is a directory and must a regular file.",
+                    filename
+                )
+            }
+            Box::new(File::create(output).unwrap_or_else(|_| {
+                panic!("Cannot create the file {} to dump coverage data", filename)
+            }))
+        }
```

Comment:
It'd be nice to do this directly in main, so we don't waste time parsing everything and then error out at the end, but this is OK too.